src/main/java/uk/gov/pay/directdebit/payments/resources/PaymentRequestResource.java

```diff
@@ -16,70 +10,77 @@
+    public PaymentRequestResource(TransactionService transactionService,
+            MandateService mandateService) {
+        this.transactionService = transactionService;
+        this.mandateService = mandateService;
     }
 
     @GET
     @Path(CHARGE_API_PATH)
     @Produces(APPLICATION_JSON)
-    public Response getCharge(@PathParam("accountId") String accountExternalId, @PathParam("paymentRequestExternalId") String paymentRequestExternalId, @Context UriInfo uriInfo) {
-        PaymentRequestResponse response = paymentRequestService.getPaymentWithExternalId(accountExternalId, paymentRequestExternalId, uriInfo);
+    public Response getCharge(@PathParam("accountId") String accountExternalId, @PathParam("transactionExternalId") String transactionExternalId, @Context UriInfo uriInfo) {
+        PaymentRequestResponse response = transactionService.getPaymentWithExternalId(accountExternalId, transactionExternalId, uriInfo);
         return Response.ok(response).build();
     }
 
     @GET
-    @Path("/v1/accounts/{accountId}/payment-requests/{paymentRequestExternalId}")
+    @Path("/v1/accounts/{accountId}/mandates/{mandateExternalId}/payments/{transactionExternalId}")
     @Produces(APPLICATION_JSON)
-    public Response getPaymentRequest(@PathParam("accountId") String accountExternalId, @PathParam("paymentRequestExternalId") String paymentRequestExternalId) {
-        LOGGER.info("Retrieving charge for frontend - {}", paymentRequestExternalId);
-        PaymentRequestFrontendResponse response = paymentRequestService.getPaymentWithExternalId(accountExternalId, paymentRequestExternalId);
+    public Response getMandateWithTransaction(@PathParam("accountId") String accountExternalId, @PathParam("mandateExternalId") String mandateExternalId, @PathParam("transactionExternalId") String transactionExternalId) {
+        LOGGER.info("Retrieving mandate {} and charge {} for frontend", mandateExternalId, transactionExternalId);
+        Mandate mandate = mandateService.findByExternalId(mandateExternalId);
+        Transaction transaction = transactionService.findTransactionForExternalIdAndGatewayAccountExternalId(transactionExternalId, accountExternalId);
+        PaymentRequestFrontendResponse response = transactionService.populateFrontendResponse(accountExternalId, mandate, transaction);
+        return Response.ok(response).build();
+    }
+
+    @GET
+    @Path("/v1/accounts/{accountId}/mandates/{mandateExternalId}")
+    @Produces(APPLICATION_JSON)
+    public Response getMandateWithoutTransaction(@PathParam("accountId") String accountExternalId, @PathParam("mandateExternalId") String mandateExternalId) {
+        LOGGER.info("Retrieving mandate {} for frontend", mandateExternalId);
+        Mandate mandate = mandateService.findByExternalId(mandateExternalId);
+        PaymentRequestFrontendResponse response = transactionService.populateFrontendResponse(accountExternalId, mandate, null);
```

Comment:
This could (and should) probably go in the mandate service, let me check if there's a reason I didn't do it in the first place

Reply:
oh my, just spotted a whole unfinished piece of work. I've moved a few things (get mandate) in the mandate resource, instead of leaving it in the payment request resource